yarn-project/aztec.js/src/utils/secrets.ts

```diff
@@ -0,0 +1,12 @@
+import { CircuitsWasm, Fr } from '@aztec/circuits.js';
+import { computeSecretMessageHash } from '@aztec/circuits.js/abis';
+
+/**
+ * Given a secret, it computes its pederson hash - used to send l1 to l2 messages
+ * @param secret - the secret to hash (defaults to a random field element)
```

Comment:
Don't like the default random value. If you don't pass something in, you won't know the secret and the message will be impossible to consume later unless you bruteforce the pre-image. If you want the random case, should also be returning the secret and not just the hash.
 
![](https://media.tenor.com/O1XNX65GPqYAAAAM/cant-reach-so-close.gif)